prompterator/utils.py

```diff
@@ -362,3 +380,47 @@ def insert_hidden_html_marker(helper_element_id, target_streamlit_element=None):
 def format_traceback_for_markdown(text):
     text = re.sub(r" ", "&nbsp;", text)
     return re.sub(r"\n", "\n\n", text)
+
+
+def normalize_name(name: str):
+    return re.sub("[^a-zA-Z0-9_]", "_", name).lower()
```

Comment:
@gkaretka shouldn't this throw an exception rather than normalize?

Reply:
This is a double edge sword since many of the json schemas you can find online do not follow required json schema for _function calling_. Eg. here: https://json-schema.org/learn/miscellaneous-examples schema `Enumerated Values` so it's up to us to decide, what do you think?

Reply:
@gkaretka my personal preference would be to be explicit here and require the _function calling_ names if possible.